File: smartcare_mobile/lib/models/doctor.dart
class DoctorModel {
  final String id;
  final String doctorName;
  final String specialization;
  final String departmentId;
  final String departmentName;
  final String branchId;
  final String branchName;
  final String roomNumber;
  final double consultationFee;
  final int avgConsultationMinutes;
  final String qualification;
  final String experienceYears;

  DoctorModel({
    required this.id,
    required this.doctorName,
    required this.specialization,
    required this.departmentId,
    required this.departmentName,
    required this.branchId,
    required this.branchName,
    required this.roomNumber,
    required this.consultationFee,
    required this.avgConsultationMinutes,
    required this.qualification,
    required this.experienceYears,
  });

  factory DoctorModel.fromJson(Map<String, dynamic> json) {
    return DoctorModel(
      id: json['id'] ?? '',
      doctorName: json['doctorName'] ?? '',
      specialization: json['specialization'] ?? '',
      departmentId: json['departmentId'] ?? '',
      departmentName: json['departmentName'] ?? '',
      branchId: json['branchId'] ?? '',
      branchName: json['branchName'] ?? '',
      roomNumber: json['roomNumber'] ?? '',
      consultationFee: (json['consultationFee'] ?? 0.0).toDouble(),
      avgConsultationMinutes: json['avgConsultationMinutes'] ?? 15,
      qualification: json['qualification'] ?? 'MD',
      experienceYears: json['experienceYears'] ?? '10 yrs',
    );
  }
}

File: smartcare_backend/src/main/java/com/smartcare/model/Doctor.java
package com.smartcare.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "doctors")
public class Doctor {
    @Id
    private String id;
    private String userId;
    private String doctorName;
    private String specialization;
    private String departmentId;
    private String departmentName;
    private String branchId;
    private String branchName;
    private String roomNumber;
    private double consultationFee;
    @Builder.Default
    private int avgConsultationMinutes = 15;
    private String qualification;
    private String experienceYears;
    private boolean active;
}
